packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -1507,6 +1516,737 @@ describe('paybackloan for dusd using dfi', () => {
+    expect(burnInfoBefore.paybacktokens).toStrictEqual([])
+    await testing.generate(4)
+
+    const tslaPaybackAmount = 1
+    const btcPaybackAmount = 1
+    const paybackLoanBlockHeight = await testing.rpc.blockchain.getBlockCount()
+    await testing.rpc.loan.paybackLoan({
+      vaultId: vaultId,
+      from: vaultOwnerAddress,
+      loans: [{
+        dToken: dusdId,
+        amounts: [`${tslaPaybackAmount}@TSLA`, `${btcPaybackAmount}@BTC`]
+      }]
+    })
+
+    // Price of tsla to dusd depends on the oracle, in this case 1 TSLA = 1000 DUSD
+    const effectiveDusdPerTsla = new BigNumber(1000).multipliedBy(1 - tslaPenaltyRate) // (DUSD per TSLA * (1 - penalty rate))
+    const dusdPaybackByTsla = new BigNumber(tslaPaybackAmount).multipliedBy(effectiveDusdPerTsla)
+
+    // Price of btc to dusd depends on the oracle, in this case 1 BTC = 1000 DUSD
+    const effectiveDusdPerBtc = new BigNumber(1000).multipliedBy(1 - btcPenaltyRate) // (DUSD per BTC * (1 - penalty rate))
+    const dusdPaybackByBtc = new BigNumber(btcPaybackAmount).multipliedBy(effectiveDusdPerBtc)
+
+    const dusdPayback = dusdPaybackByBtc.plus(dusdPaybackByTsla)
+
+    const dusdLoanRemainingAfter = new BigNumber(dusdLoanAmount).minus(dusdPayback).plus(dusdInterestAmountBefore)
+    const dusdInterestPerBlockAfter = dusdLoanRemainingAfter.multipliedBy(netInterest).dividedBy(365 * blocksPerDay)
+
+    // Let some time, generate blocks
+    await testing.generate(2)
+
+    const totalTslaPenalty = new BigNumber(tslaPaybackAmount).multipliedBy(tslaPenaltyRate)
+    const totalBtcPenalty = new BigNumber(btcPaybackAmount).multipliedBy(btcPenaltyRate)
+    const burnInfoAfter = await testing.rpc.account.getBurnInfo()
+    expect(burnInfoAfter.tokens).toStrictEqual([`${new BigNumber(tslaPaybackAmount).toFixed(8)}@TSLA`, `${new BigNumber(btcPaybackAmount).toFixed(8)}@BTC`])
+    expect(burnInfoAfter.dfipaybackfee).toStrictEqual(new BigNumber(0))
+    expect(burnInfoAfter.dfipaybacktokens).toStrictEqual([])
+    expect(burnInfoAfter.paybackfees).toStrictEqual([`${totalTslaPenalty.toFixed(8)}@TSLA`, `${totalBtcPenalty.toFixed(8)}@BTC`])
+    expect(burnInfoAfter.paybacktokens).toStrictEqual([`${dusdPayback.toFixed(8)}@DUSD`])
+
+    const blockHeightAfter = await testing.rpc.blockchain.getBlockCount()
+    const dusdInterestAmountAfter = dusdInterestPerBlockAfter.multipliedBy(blockHeightAfter - paybackLoanBlockHeight).decimalPlaces(8, BigNumber.ROUND_CEIL)
+    const dusdLoanAmountAfter = dusdLoanRemainingAfter.plus(dusdInterestAmountAfter)
+
+    const vaultAfter = await testing.rpc.loan.getVault(vaultId) as VaultActive
+    // expect(vaultAfter.loanAmounts).toStrictEqual([`${dusdLoanAmountAfter.toFixed(8)}@DUSD`])
+    // expect(vaultAfter.interestAmounts).toStrictEqual([`${dusdInterestAmountAfter.toFixed(8)}@DUSD`])
+    console.log(`dusdLoanAmountAfter - from vault ${vaultAfter.loanAmounts[0]}, from calculation ${dusdLoanAmountAfter.toFixed(8)}@DUSD`)
```

Comment:
Need help, here vault information received are not matched with calculations. below are the results.
```
dusdLoanAmountAfter - from vault 3030.04055941@DUSD, from calculation 3030.02914388@DUSD
dusdInterestAmountAfter- from vault 0.00345895@DUSD, from calculation 0.00345894@DUSD 
```

Reply:
Could be L1883 `vaultAfter` do include 1 (current) extra block of interest?
`const dusdInterestAmountAfter = dusdInterestPerBlockAfter.multipliedBy(blockHeightAfter - paybackLoanBlockHeight + 1).decimalPlaces(8, BigNumber.ROUND_CEIL)`

Reply:
Thanks Ivan for looking, Reason was extra blocks generated at L1843, This is resolved.